impala/dbapi/hiveserver2.py

```diff
@@ -201,13 +201,49 @@ def _get_sleep_interval(self, start_time):
             return 0.5
         return 1.0
 
-    def executemany(self, operation, seq_of_parameters):
+    def _exec_single_query_insert(self, operation, seq_of_parameters):
+        """
+        Builds and executes one big INSERT...VALUES statement instead of sending
+        sequency of one-row inserts.
+        Assumes that `operation` is INSERT...VALUES statement.
+        Returns `True` if operation was executed and `False` otherwise.
+
+        INSERT...VALUES is available in Impala and multiple row inserts are
+        supported.
+        """
+        match = RE_INSERT_VALUES.match(operation)
+        if match and seq_of_parameters and len(seq_of_parameters) > 1:
+            # Split "INSERT...VALUES (...)" query into
+            query_left = match.group(1)  # "INSERT...VALUES"
+            values = match.group(2)      # "(...)"
+
+            def op():
+                # Bind each row's parameters to `values` [ "(...)" ]
+                bound_params = map(
+                    lambda params: _bind_parameters(values, params),
+                    seq_of_parameters)
+                # Build "big" query: "INSERT...VALUES (...), (...), (...), ..."
+                self._last_operation_string = query_left + ", ".join(bound_params)
+                self._last_operation_handle = rpc.execute_statement(
+                    self.service, self.session_handle, self._last_operation_string,
+                    None)
+
+            self._execute_sync(op)
+            return True
+        return False
+
+    def executemany(self, operation, seq_of_parameters, single_query_insert=False):
```

Comment:
Perhaps change the name of the new flag to `rewrite_inserts_single_query`, or something like that?  I think it would be more clear as to exactly what's happening.